### PR TITLE
feat: introduce oa v4 (alpha) context and schema

### PR DESCRIPTION
**Pull request by HJunyuan**
## What does this PR do?
- Rename existing OA v3 contexts to be categorised under `3.0`
- Introduce OA v4 (alpha) context and schema categorised under `4.0`

### CODE REVIEW

**Comment by cavacado** [comment]
hmm, all existing v3 docs will soft break immediately though, since it wont be able find the context.
<BLAH>/com/openattestation/1.0/OpenAttestation.v3.json
-> <BLAH>/com/openattestation/3.0/OpenAttestation.v3.json

**Comment by HJunyuan** [comment]
> hmm, all existing v3 docs will soft break immediately though, since it wont be able find the context. PATH/com/openattestation/1.0/OpenAttestation.v3.json -> PATH/com/openattestation/3.0/OpenAttestation.v3.json

@cavacado Oh yes, thanks for pointing that out. Not sure why it was categorised as `1.0` from the beginning 🤔

Should we revert to `1.0`? Or we can [create a redirect](https://github.com/Open-Attestation/schemata/blob/master/_redirects) for backwards compatibility?

**Comment by cavacado** [comment]
redirect better i guess, since your dir struct correctly reflects the intent.

**Comment by HJunyuan** [comment]
@cavacado Redirects have been implemented.

Can try it out in the deploy preview link: https://deploy-preview-78--open-attestation-schemata.netlify.app/com/openattestation/1.0/OpenAttestation.v3.json

File: src/com/openattestation/4.0/context.json
Comment (HJunyuan) [inline src/com/openattestation/4.0/context.json:1]: Also do we have any strong preference on the file name? Considering it is quite permanent.

Could be one of these:
- `context.json`
- `OpenAttestationCredentialContext.json`
- etc.

For reference:
- W3C uses this URL: https://www.w3.org/2018/credentials/v1
- W3C-CCG uses this URL: https://w3id.org/traceability/v1 but it gets redirected to https://w3c-ccg.github.io/traceability-vocab/contexts/traceability-v1.jsonld

**Comment by HJunyuan** [comment]
@cavacado Shall we merge this in?

I'm updating the `open-attestation` library at the moment and it would need to prefetch the OA v4 context: `https://schemata.openattestation.com/com/openattestation/4.0/alpha-context.json`

(Similar behaviour to the OA v3 contexts as well)

File: src/com/openattestation/4.0/alpha-schema.json [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+      "required": ["renderMethodType", "name", "url"]
+    },
+    "credentialSubject": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "format": "uri",
+          "description": "URI when dereferenced, results in a document containing machine-readable information about the issuer that can be used to verify the information expressed in the credential. More information in https://www.w3.org/TR/vc-data-model/#issuer"
+        },
```
Comment (cavacado) [inline src/com/openattestation/4.0/alpha-schema.json:142]: the description might not be correct, i thought its the id of the object of claim
https://www.w3.org/TR/vc-data-model-2.0/#credential-subject

**Review by cavacado — COMMENTED** [review]
check on the comment? but other than that ok lets merge this alpha piece.

**Review by cavacado — APPROVED** [review]

File: src/com/openattestation/4.0/alpha-schema.json (hunk 1 above)
Comment (HJunyuan) [reply src/com/openattestation/4.0/alpha-schema.json:142]: Oh yes, good catch. Looking again at the [documentation](https://www.w3.org/TR/vc-data-model/#credential-subject), we can reword it to:

> An identifier related to the subject of the verifiable credential. More information in https://www.w3.org/TR/vc-data-model/#credential-subject
